Code_Exercises/9_sycl_of_hell/README.md

```diff
@@ -0,0 +1,34 @@
+![Sandro_Botticelli_-_La_Carte_de_l'Enfer](https://github.com/user-attachments/assets/f657847c-1e53-49d7-b16d-33701989b812)
+
+# How to compile
+
+## Goold Old Make
+
+```
+# Aurora
+CXX=icpx CXXFLAGS=-fsycl make -j
+# Polaris
+module use /soft/modulefiles
+module load oneapi/upstream
+CXX=clang++ CXXFLAGS="-fsycl -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_80" make -j
```

Comment:
I had set up an AdaptiveCpp installation on Polaris for the IWOCL tutorial. If there's interest I could see if that still works.

Reply:
Okay, will try Polaris tomorrow.

We always had working AdaptiveCpp in SYCLAcademy cmake. I think the only thing that would need to change in your `CMakeLists.txt` is a `find_package(AdaptiveCpp)`.
Not sure if it's an issue if `add_sycl_to_target` is invoked on every source file (usually it's invoked once per target with a list of source files), but probably it will work.
AdaptiveCpp already ships this cmake interface.

Reply:
>Not sure if it's an issue if add_sycl_to_target is invoked on every source file (usually it's invoked once per target with a list of source files), but probably it will work.

Each `.cpp` are independent mains, so can use the `foreach` hack :) 

>We always had working AdaptiveCpp in SYCLAcademy cmake. I think the only thing that would need to change in your CMakeLists.txt is a find_package(AdaptiveCpp).

Oh then perfect. Can try on Polaris with Adaptive cpp (when you give me the green light) and move to the  SYCLAcamedy CMake away

Reply:
Ah, I missed that these are multiple targets, one for each file. Then all should be fine and you just need `find_package(AdaptiveCpp)`.

Will let you know once I've tested on Polaris.

Reply:
```
find_package(IntelSYCL QUIET)
IF(NOT IntelSYCL_FOUND )
	find_package(AdaptiveCpp REQUIRED)
ENDIF()
```
Best i can do, 
When what I want is 
```
if not(any(find_package(foo) for foo in ["IntelSYCL","IntelSYCL"])
   sys.exit("No SYCK package found")
```
But good enough I assume.

Reply:
Something like this should do this:

```
find_package(IntelSYCL QUIET)
find_package(AdaptiveCpp QUIET)
if(NOT IntelSYCL_FOUND AND NOT AdaptiveCpp_FOUND)
  message(SEND_ERROR "No SYCL implementation!")
endif()
# Perhaps also think about case when both are available?
```

But I guess such polishing can always be done at a later point :)